### PR TITLE
[Bug-fix] change char* to unsigned char* in vec/function/string

### PR DESCRIPTION
**Pull request by zbtzbtzbt**
## Proposed changes
change char* to unsigned char*.

the following changes are only related to vec/function/string in this pr.
in pr [fix-reverse](https://github.com/doris-vectorized/doris-vectorized/pull/142),i find use char* to operate string will cause wrong ans in some cases
such as:

```
mysql> select reverse('美团和和阿斯顿ab');
+------------------------------------+
| reverse('美团和和阿斯顿ab')        |
+------------------------------------+
| �ab�顿斘�和和团美                     |
+------------------------------------+
1 row in set (0.01 sec)
```
so I change related char* to unsigned char* in vec function string.

detail : [difference of char* 和 unsigned char* in C](https://blog.csdn.net/guotianqing/article/details/77341657?ops_request_misc=%257B%2522request%255Fid%2522%253A%2522163058273016780264073154%2522%252C%2522scm%2522%253A%252220140713.130102334.pc%255Fblog.%2522%257D&request_id=163058273016780264073154&biz_id=0&utm_medium=distribute.pc_search_result.none-task-blog-2~blog~first_rank_v2~rank_v29-1-77341657.pc_v2_rank_blog_default&utm_term=char%E5%92%8C&spm=1018.2226.3001.4450)

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/function_string.cpp [hunk 1]
```diff
@@ -129,8 +129,8 @@ struct EndsWithOp {
     using ResultPaddedPODArray = PaddedPODArray<Int8>;
 
     static void execute(const std::string_view& strl, const std::string_view& strr, int8_t& res) {
-        re2::StringPiece str_sp(const_cast<char*>(strl.data()), strl.length());
```
Comment (zuochunwei) [inline be/src/vec/functions/function_string.cpp:132]: 此处不需要const_cast
函数接受const char* p，那表示这个函数向调用者承诺，放心大胆的调用吧，我（函数实现）不会修改p指向的内存。你传const char* p1, 传char* p2都ok。

如果函数设计者，这样设计函数：
void func(char* p);

那么你下面调用会失败
const char* p = "good";
func(p);

因为你的func没有承诺不会改p指向的内存，但传递的p是readonly常量，函数的承诺不够，所以编译会报错，其实是编译器帮我们做安全检查。

建议你多了解一下const，比较系统的掌握它

File: be/src/vec/functions/function_bitmap.cpp [hunk 2]
```diff
@@ -50,7 +50,7 @@ struct ToBitmapImpl {
         auto size = offsets.size();
         res.reserve(size);
         for (int i = 0; i < size; ++i) {
-            const char* raw_str = reinterpret_cast<const char*>(&data[offsets[i - 1]]);
```
Comment (zuochunwei) [inline be/src/vec/functions/function_bitmap.cpp:53]: 为什么改成unsigned char，没懂

File: be/src/vec/functions/function_string.cpp (hunk 1 above)
Comment (zbtzbtzbt) [reply be/src/vec/functions/function_string.cpp:132]: 我明白了，感谢

File: be/src/vec/functions/function_bitmap.cpp (hunk 2 above)
Comment (zbtzbtzbt) [reply be/src/vec/functions/function_bitmap.cpp:53]: 为什么改成unsigned char后，中文字符就不会乱码了呢？
比如说现在有个中文字符 '左'，大小是3个字节，
假设啊，假设它的二进制表示是 10110110 00110010 00010011，
在be层，如果用char*表示，在上推到fe层时，可能就是int了，
在char*转int时，如果最高位是0，那么没问题，用0拓展，这种情况下不会出错；如果最高位是1，会进行有符号拓展，高位都补1，转int后，会变成负数。而unsigned char*只会进行无符号拓展，补0，始终是正数。
目前需要补0的逻辑才是对的，这样中文字符才不会出现乱码。

**Pull request closed without merging**